packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/tile_overlay_updates.dart

```diff
@@ -0,0 +1,125 @@
+    }
+
+    final Map<TileOverlayId, TileOverlay> previousTileOverlays =
+        keyTileOverlayId(previous);
+    final Map<TileOverlayId, TileOverlay> currentTileOverlays =
+        keyTileOverlayId(current);
+
+    final Set<TileOverlayId> prevTileOverlayIds =
+        previousTileOverlays.keys.toSet();
+    final Set<TileOverlayId> currentTileOverlayIds =
+        currentTileOverlays.keys.toSet();
+
+    TileOverlay idToCurrentTileOverlay(TileOverlayId id) {
+      return currentTileOverlays[id];
+    }
+
+    _tileOverlayIdsToRemove =
+        prevTileOverlayIds.difference(currentTileOverlayIds);
+
+    _tileOverlaysToAdd = currentTileOverlayIds
+        .difference(prevTileOverlayIds)
+        .map(idToCurrentTileOverlay)
+        .toSet();
+
+    // Returns `true` if [current] is not equals to previous one with the
+    // same id.
+    bool hasChanged(TileOverlay current) {
+      final TileOverlay previous = previousTileOverlays[current.tileOverlayId];
+      return current != previous;
+    }
+
+    _tileOverlaysToChange = currentTileOverlayIds
+        .intersection(prevTileOverlayIds)
+        .map(idToCurrentTileOverlay)
+        .where(hasChanged)
+        .toSet();
+  }
+
+  /// Set of TileOverlays to be added in this update.
+  Set<TileOverlay> get tileOverlaysToAdd {
+    return _tileOverlaysToAdd;
+  }
+
+  Set<TileOverlay> _tileOverlaysToAdd;
+
+  /// Set of TileOverlayIds to be removed in this update.
+  Set<TileOverlayId> get tileOverlayIdsToRemove {
+    return _tileOverlayIdsToRemove;
+  }
+
+  Set<TileOverlayId> _tileOverlayIdsToRemove;
+
+  /// Set of TileOverlays to be changed in this update.
+  Set<TileOverlay> get tileOverlaysToChange {
+    return _tileOverlaysToChange;
+  }
+
+  Set<TileOverlay> _tileOverlaysToChange;
+
+  /// Converts this object to JSON.
+  Map<String, dynamic> toJson() {
+    final Map<String, dynamic> updateMap = <String, dynamic>{};
+
+    void addIfNonNull(String fieldName, dynamic value) {
+      if (value != null) {
+        updateMap[fieldName] = value;
+      }
+    }
+
+    addIfNonNull(
+        'tileOverlaysToAdd', serializeTileOverlaySet(_tileOverlaysToAdd));
+    addIfNonNull(
+        'tileOverlaysToChange', serializeTileOverlaySet(_tileOverlaysToChange));
+    addIfNonNull(
+        'tileOverlayIdsToRemove',
+        _tileOverlayIdsToRemove
+            .map<dynamic>((TileOverlayId m) => m.value)
+            .toList());
+
+    return updateMap;
+  }
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final TileOverlayUpdates typedOther = other;
+    return setEquals(_tileOverlaysToAdd, typedOther._tileOverlaysToAdd) &&
+        setEquals(
+            _tileOverlayIdsToRemove, typedOther._tileOverlayIdsToRemove) &&
+        setEquals(_tileOverlaysToChange, typedOther._tileOverlaysToChange);
+  }
+
+  @override
+  int get hashCode =>
```

Comment:
Use hashValues (https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#common-boilerplates-for-operator--and-hashcode)

Reply:
done